src/app/code/community/Hackathon/PromoCodeMessages/Model/Validator.php

```diff
@@ -170,6 +170,15 @@ protected function _validateGeneral($rule, $coupon)
             }
         }
 
+        // magemail coupon-level auto-expiration date
+        if ($coupon->getdata('magemail_expired_at') && $coupon->getUsageLimit() && $coupon->getTimesUsed() >= $coupon->getUsageLimit()) {
```

Comment:
@philwinkle broke the line.

As far as the `getData()` goes, I usually prefer the getter method if it's available as an actual method or docblock.  Otherwise it throws a warning in the IDE so I usually prefer the plain `getData()`.

Not sure what your thoughts are on that @lfolco ?